include/highfive/H5Easy.hpp

```diff
@@ -39,20 +39,151 @@
 
 namespace H5Easy {
 
+using HighFive::Attribute;
 using HighFive::AtomicType;
 using HighFive::Chunking;
 using HighFive::DataSet;
 using HighFive::DataSetCreateProps;
 using HighFive::DataSpace;
+using HighFive::Deflate;
 using HighFive::Exception;
 using HighFive::File;
 using HighFive::ObjectType;
+using HighFive::Shuffle;
 
 ///
 /// \brief Write mode for DataSets
 enum class DumpMode {
-    Create, /*!< Dump only if DataSet does not exist, otherwise throw. */
-    Overwrite /*!< If DataSet already exists overwrite it if data has the same shape, otherwise throw. */
+    Create = 0, /*!< Dump only if DataSet does not exist, otherwise throw. */
+    Overwrite = 1 /*!< Create or overwrite if DataSet of correct shape exists, otherwise throw. */
+};
+
+///
+/// \brief Enable/disable automatic flushing after write operations.
+enum class Flush
+{
+    False = 0, /*!< No automatic flushing. */
+    True = 1 /*!< Automatic flushing. */
+};
+
+///
+/// \brief Enable/disable compression for written DataSets.
+enum class Compression
+{
+    False = 0, /*!< No compression. */
+    Medium = 5, /*!< Medium compression (deflate level 5). */
+    High = 9 /*!< High compression (deflate level 9). */
+};
+
+///
+/// \brief Set compression-level for written DataSets.
+class CompressionLevel
```

Comment:
I'm wondering if this and `Compression` could not be folded into one, simpler class/enum; Numerical values count for compression level, certain values could be queried by i.e.:
```
bool compress() const { return m_level > 0; };
```
etc…

You could use a static construction method to get medium/high compression.

Reply:
This I like. A question is if `deflate_level = 0` means no compression, also in the underlying library? Or if the user should be able to really specify `deflate_level = 0`.

Reply:
I get the answer is `deflate_level = 0` should switch of shuffling as well: https://github.com/Unidata/netcdf-c/issues/23

Reply:
About converting to one class, I love the idea, but I don't think it can be done though. C++ does not have static constructors, and having `enum` members would ruin the overload resolution.

Unless I'm overlooking something @matz-e ?

Reply:
Just an idea or two:
```
#include <iostream>

class SomeLevel {
    size_t level_;
public:
    SomeLevel(size_t l) : level_(l) {};

    static const SomeLevel High;
    static SomeLevel Medium() {
        return SomeLevel{4};
    };

    size_t get() const { return level_; };
};

const SomeLevel SomeLevel::High{8};

int main(void) {
    auto foo = SomeLevel::High;
    std::cout << foo.get() << "\n";
    auto bar = SomeLevel::Medium();
    std::cout << bar.get() << "\n";
}
```

If there should really be a distinction between 0 and not setting any compression, I would think that -1/limits::max<unsigned> could be used?

Reply:
Beautiful! Thanks a lot!

Reply:
This really simplifies not only the implementation, but also the API!

Reply:
@matz-e This leads to a a compiler error "multiple definition of ". Do you know if and how that can be solved for a header-only library?